charts/replicated-library/templates/lib/_statefulset.tpl

```diff
@@ -30,14 +40,15 @@ spec:
   {{- end }}
   updateStrategy:
     type: {{ $strategy }}
-    {{- if and (eq $strategy "RollingUpdate") $values.rollingUpdate.partition}}
+    {{- if and (eq $strategy "RollingUpdate") (and $values.rollingUpdate $values.rollingUpdate.partition) }}
     rollingUpdate:
       partition: {{ $values.rollingUpdate.partition }}
     {{- end }}
   selector:
     matchLabels:
       {{- include "replicated-library.labels.selectorLabels" . | nindent 6 }}
-  serviceName:  {{- default (include "replicated-library.names.appname" .) ($values.serviceName) }}
+  serviceName: {{ default (include "replicated-library.firstservice" .) ($values.serviceName) }}
```

Comment:
if `$values.serviceName` isn't set and there is not matching service found via your function, will this set `serviceName` to `[]` and will that cause any issues?

Reply:
That's a good questions, yes I do think it's going to return an empty list and no I don't know if that's ok or not. I'll test it.